app/controllers/spree/admin/invoices_controller.rb

```diff
@@ -23,6 +23,12 @@ def generate
         OrderInvoiceGenerator.new(@order).generate_or_update_latest_invoice
         redirect_back(fallback_location: spree.admin_dashboard_path)
       end
+
+      private
+
+      def filepath(invoice_id)
+        "tmp/invoices/#{invoice_id}.pdf"
+      end
```

Comment:
This looks bad. Not new in this PR but it does look like I could download any PDF from the server. That said, my tests show that Rails is clever enough to not recognise the route if I inject `..%2F..%2Fdoc%2Ferd` as invoice id. So we are good for now. Still, it would be good to use Active Storage for better security.